src/types/resolved.ts

```diff
@@ -8,4 +9,6 @@ export type ResolvedRoute<TRoute extends Route = Route> = {
   key: TRoute['key'],
   query: ResolvedRouteQuery,
   params: ExtractRouteParamTypes<TRoute>,
-}
\ No newline at end of file
+}
+
+export type ReadonlyResolvedRoute<TRoute extends ResolvedRoute = ResolvedRoute> = DeepReadonly<Omit<TRoute, 'params'>> & Readonly<{ params: TRoute['params'] }>
```

Comment:
I'm not sold on 
1. Introducing another route type
2. The name.

Reply:
Should ResolvedRoute be read only by default?

Reply:
Possibly? Once we resolve a route do we ever need to update it internally?

Reply:
I think that's fine, I would proceed with that assumption and we can pivot in the future if we need to